observer/probers/tls/tls.go

```diff
@@ -0,0 +1,165 @@
+package probers
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/crypto/ocsp"
+	"io"
+	"net/http"
+	"time"
+)
+
+var StatusCode int
+
+const (
+	Success = iota
+	UnexpectedRoot
+	UnexpectedResponse
+	UnexpectedRootAndResponse
+	UnknownOCSPCheck
+	FailedOCSPCheck
+	FailedTLSConnection
+	Unknown
+)
+
+// TLSProbe is the exported 'Prober' object for monitors configured to
+// perform TLS protocols.
+type TLSProbe struct {
+	url        string
+	root       string
+	response   string
+	certExpiry *prometheus.GaugeVec
+	StatusCode *prometheus.GaugeVec
+}
+
+// Name returns a string that uniquely identifies the monitor.
+func (p TLSProbe) Name() string {
+	return fmt.Sprintf("%s--expecting-%s-%s", p.url, p.response, p.root)
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Prober`.
+func (p TLSProbe) Kind() string {
+	return "TLS"
+}
+
+func isOCSPRevoked(cert, issuer *x509.Certificate) bool {
+	req, err := ocsp.CreateRequest(cert, issuer, nil)
+	if err != nil {
+		StatusCode = FailedOCSPCheck
+		return false
```

Comment:
I think the return for this function should probably be `error`. In instances where we actually have an underlying error we can simply return that error. In instances where we have a value other than `ocsp.Revoked` we can return a package level error that we can check for in `.Probe()`